class-04-templates/pair-assignment/sam_pab/scripts/article.js

```diff
@@ -0,0 +1,36 @@
+var articles = [];
+
+function Article (opts) {
+  this.author = opts.author;
+  this.authorUrl = opts.authorUrl;
+  this.title = opts.title;
+  this.category = opts.category;
+  this.body = opts.body;
+  this.publishedOn = opts.publishedOn;
+}
+
+Article.prototype.toHtml = function() {
+  var articleTemplate = $('#article-template').html();
+  var compileTemplate = Handlebars.compile(articleTemplate);
+
+  // DONE: If your template will use properties that aren't on the object yet, add them.
+  //   Since your template can't hold any JS logic, we need to execute the logic here.
+  //   The result is added to the object as a new property, which can then be referenced by key in the template.
+  //   For example, you might want to display how old a post is, or say "(draft)" if it has no publication date:
+  this.daysAgo = parseInt((new Date() - new Date(this.publishedOn))/60/60/24/1000);
+  this.publishStatus = this.publishedOn ? 'published ' + this.daysAgo + ' days ago' : '(draft)';
+  var html = compileTemplate(this);
+  $('#articles').append(html);
```

Comment:
some duplication here with line 35. You only need to append once. In the .toHtml function, you can simply ```return html``` and then line 35 will append each article in the article array.